### PR TITLE
New errors structure

### PR DESCRIPTION
**Pull request by mgordel**
This PR contains a lot of changes in places where errors were thrown and their handling. The structure of new errors has changed according to the adopted assumptions

### CODE REVIEW

File: tests/mock/rest/activity.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
 import { v4 as uuidv4 } from "uuid";
 import { ExeScriptCommandResultResultEnum } from "ya-ts-client/dist/ya-activity/src/models/exe-script-command-result";
 import { RequestorStateApi } from "ya-ts-client/dist/ya-activity/src/api/requestor-state-api";
+import { Result } from "../../../src";
```
Comment (SewerynKras) [inline tests/mock/rest/activity.ts:15]: unused import?

**Review by SewerynKras — COMMENTED** [review]
Excellent job, I can't promise that I tested all possible scenarios, but this looks really, really promising. I left a couple small observations

File: src/market/demand.ts [hunk 2]
```diff
@@ -223,15 +222,25 @@ export class Demand extends EventTarget {
         if (this.isRunning) {
           const reason = error.response?.data?.message || error;
           this.options.eventTarget?.dispatchEvent(new Events.CollectFailed({ id: this.id, reason }));
-          this.logger.warn(`Unable to collect offers.`, { reason });
-          if (error.code === "ECONNREFUSED" || error.response?.status === 404) {
+          this.logger.warn(`Unable to collect offers. ${reason}`);
```
[the remark this answers is not in the document]
Comment (mgordel) [reply src/market/demand.ts:225]: I changed it by mistake

File: src/market/service.ts [hunk 3]
```diff
@@ -93,9 +98,14 @@ export class MarketService {
...
+    if (error instanceof GolemMarketError && error.code === MarketErrorCode.DemandExpired) {
+      this.logger.error("Demand expired. Trying to subscribe a new one...");
+      this.resubscribeDemand().catch((e) => this.logger?.warn(e));
+      return;
+    }
     if (error || !proposal) {
-      this.logger.warn("Subscription failed. Trying to subscribe a new one...");
-      this.resubscribeDemand().catch((e) => this.logger.warn(`Could not resubscribe demand.`, e));
+      this.logger.error("Collecting offers failed. Trying to subscribe a new demand...");
+      this.resubscribeDemand().catch((e) => this.logger?.warn(e));
```
[the remark this answers is not in the document]
Comment (mgordel) [reply src/market/service.ts:108]: the same - by mistake

File: src/task/process.ts [hunk 4]
```diff
@@ -36,15 +40,33 @@ export class RemoteProcess {
     return new Promise((resolve, reject) => {
       const timeoutInMs = timeout ?? DEFAULTS.exitWaitingTimeout;
       const timeoutId = setTimeout(
-        () => reject(new GolemError(`The waiting time (${timeoutInMs} ms) for the final result has been exceeded`)),
+        () =>
+          reject(
+            new GolemWorkError(
+              `todo`,
```
[the remark this answers is not in the document]
Comment (mgordel) [reply src/task/process.ts:46]: :)

File: src/executor/config.ts [hunk 5]
```diff
@@ -52,10 +52,10 @@ export class ExecutorConfig {
     this.activityExecuteTimeout = options.activityExecuteTimeout || options.taskTimeout;
     const apiKey = options?.yagnaOptions?.apiKey || processEnv.env.YAGNA_APPKEY;
     if (!apiKey) {
-      throw new GolemError("Api key not defined");
+      throw new GolemUserError("Api key not defined");
     }
     if (options.maxTaskRetries && options.maxTaskRetries < 0) {
-      throw new GolemError("The maxTaskRetries parameter cannot be less than zero");
+      throw new GolemUserError("The maxTaskRetries parameter cannot be less than zero");
```
[the remark this answers is not in the document]
Comment (grisha87) [reply src/executor/config.ts]: I like the idea of renaming this, one suggestion is to make it shorter: `GolemConfigError`.

**Review by SewerynKras — APPROVED** [review]
LGTM

**Pull request merged**